File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */

var longestCommonPrefix = function(strs) {
    let prefix = ""
    
    if(strs.length === 0){return prefix}

    strs.sort()
    let short = strs[0]
    let long = strs[strs.length-1]

    for(let i=0; i<short.length; i++){
        if(short[i] !== long[i]){
            return prefix
        } else {prefix+=short[i]}
    }

    return prefix
};

File: 14-longest-common-prefix/longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {return ""}

    prefix := ""

    slices.Sort(strs)
    
    short := strs[0]
    long := strs[len(strs)-1]
    for i:=0; i < len(short); i++ {
        if string(short[i]) != string(long[i]){
            return prefix 
        } else {
            prefix += string(short[i])
        }
    }

    return prefix
}